src/esp_littlefs.c

```diff
@@ -899,13 +899,17 @@ static int esp_littlefs_flags_conv(int m) {
 
 static void esp_littlefs_take_efs_lock(void) {
     if( _efs_lock == NULL ){
+#if portNUM_PROCESSORS > 1  // SMP protection needed for multi-core
```

Comment:
Last issue for this PR:

Is this `portENTER_CRITICAL` still required for a single core processor? The race condition is if 2 threads attempt to initialize the global `_efs_lock`, at the same time, then without this look we could result in 2 calls of `xSemaphoreCreateMutex`. This is avoided by the seemingly-redundant `_efs_lock == NULL` check.

Reply:
Ah, good point. I found `taskENTER_CRITICAL()` and `taskEXIT_CRITICAL()` which seem to be what is needed.